src/mtev_cluster.c

```diff
@@ -131,6 +131,9 @@ struct mtev_cluster_t {
   int64_t config_seq;
   int node_cnt;
   mtev_cluster_node_t *nodes;
+  // [2018-04-10 HH] cht is a concurrent hash-table mapping (char*) => (void*).
+  // It is initialized to hold a map node-UUID-string => mtev_cluster_node_t of known cluster
+  // nodes, and is used by the methods mtev_cluster_do_i_own, mtev_cluster_filter_owners.
```

Comment:
It's not a concurrent hash table, it's a consistent hash table.  It also doesn't hold a map as you describe.  I'd simply remove this comment block.

Reply:
The cluster is represented as a consistent hash ring (like snowth).  When I have a 30 node cluster and 300k checks, which nodes runs which checks?  For pull checks only one can run them, so they need to collectively decide which nodes own which check data given the current availability.  This is what the CHT accomplishes.

Reply:
OK. I understand the intentions here, but:
a) I don't see where a mapping `node-uuid => check-run-on-that-node` is implemented and maintained. Is that TBD? or am I missing something here. The only place I can see where cht is written to is in the `mtev_cluster_compile` function linked above.
b) Am I reading this code wrong, or are we really storing a table "node-uuid-str => node struct" in the cht, and not a map "node-uuid-str => check-run-on-that-node"
```
  *(cluster->cht) = mtev_cht_alloc();
  nodes = calloc(sizeof(*nodes), cluster->node_cnt);
  for(i=0; i<cluster->node_cnt; i++) {
    char uuid_str[UUID_STR_LEN+1];
    uuid_unparse_lower(cluster->nodes[i].id, uuid_str);
    nodes[i].name = strdup(uuid_str);
    nodes[i].userdata = &cluster->nodes[i];
  }
mtev_cht_set_nodes(*(cluster->cht), cluster->node_cnt, nodes);
```